syntax = "proto3";

package protos;
option go_package = "orch-go/gen/protos;protos";
option csharp_namespace = "Core";

import "custom_types.proto";
import "google/protobuf/empty.proto";
import "login_log.proto";

service LoginLogAnalyticService {
  rpc ProcessGetAll (GetAllRequest) returns (GetAllLoginLogsResponse);
  rpc ProcessGetById (GetLoginLogByIdRequest) returns (LoginLogModel);
  rpc ProcessGetByIds (GetLoginLogByIdsRequest) returns (GetAllLoginLogsResponse);
  rpc ProcessGetByCustomer (GetLoginLogsByCustomerRequest) returns (GetAllLoginLogsResponse);
  rpc ProcessGetInTimeRange (GetLoginLogsInTimeRangeRequest) returns (GetAllLoginLogsResponse);
  rpc ProcessGetCount (google.protobuf.Empty) returns (CountResponse);
}
